aten/src/ATen/native/native_functions.yaml

```diff
@@ -1367,6 +1367,16 @@
 - func: fill_diagonal_(Tensor(a!) self, Scalar fill_value, bool wrap=False) -> Tensor(a!)
   variants: method
 
+- func: diff(Tensor self, int n=1, int dim=-1, Tensor? prepend=None, Tensor? append=None) -> Tensor
```

Comment:
cc @rgommers for a possible issue with NumPy's signature

[np.diff](https://numpy.org/doc/stable/reference/generated/numpy.diff.html?highlight=diff#numpy.diff) has both prepend and append default to "no value", which is represented by:

https://github.com/numpy/numpy/blob/1de46fe2feee9d3c500a83331ac9b75af5aef947/numpy/_globals.py#L57

The comment indicates it's intended to be used with "deprecated keywords." However, it appears these come from the PR adding the kwargs: https://github.com/numpy/numpy/pull/8206. @rgommers, seems like the NumPy signature would ideally have prepend=None and append=None?

Reply:
That comment is a little misleading, or at least incomplete. `_NoValue` is mainly used when keywords are added to a function, and the function forwards its keywords somewhere. See for example `np.sum` which forwards to the `.sum` method of its input (which can be a `torch.Tensor`). Adding keywords to the numpy function that are not yet present in all other/downstream libraries implementing the same functionality would then break previously working code. I'll go fix the code comment.

Wherever you see `_NoValue`, just read `None`.

The PR gives this example that would not work with `prepend=None` for object arrays:
```
>>> data = np.array([2, 3], dtype=object)
>>> np.diff(data, prepend=None)
array([None, 1], dtype=object)  # expected
array([1], dtype=object)  # actual
```

A little far-fetched, but unfortunately this works:
```
>>> np.array(None)                                                          
array(None, dtype=object)
```
Object arrays are very annoying.